async_grpc/client.h

```diff
@@ -71,6 +83,9 @@ class Client {
 
   bool Write(const RequestType &request, ::grpc::Status *status = nullptr) {
     ::grpc::Status internal_status;
+    if (timeout_.has_value()) {
```

Comment:
Since `Reset()` has `if (timeout_.has_value())`,  there will be 2 checks if `timeout_` has the value. Can we have only one?